File: software/examples/java/ExampleSimple.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletIndustrialQuadRelay;

public class ExampleSimple {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Industrial Quad Relay Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletIndustrialQuadRelay iqr =
		  new BrickletIndustrialQuadRelay(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Turn relays alternating on/off 10 times with 100 ms delay
		for(int i = 0; i < 10; i++) {
			Thread.sleep(100);
			iqr.setValue(1 << 0);
			Thread.sleep(100);
			iqr.setValue(1 << 1);
			Thread.sleep(100);
			iqr.setValue(1 << 2);
			Thread.sleep(100);
			iqr.setValue(1 << 3);
		}

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/industrial_quad_relay_bricklet"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
	"time"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Industrial Quad Relay Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	iqr, _ := industrial_quad_relay_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Turn relays alternating on/off 10 times with 100 ms delay
	for i := 0; i < 10; i++ {
		time.Sleep(100 * time.Millisecond)
		iqr.SetValue(1 << 0)
		time.Sleep(100 * time.Millisecond)
		iqr.SetValue(1 << 1)
		time.Sleep(100 * time.Millisecond)
		iqr.SetValue(1 << 2)
		time.Sleep(100 * time.Millisecond)
		iqr.SetValue(1 << 3)
	}

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
